### PR TITLE
Code improvements create test

### PR DESCRIPTION
**Pull request by HuziMuzi**
Feat: 
- correct display of difficulty.
- add a button to switch to a new question and exit from creating a test.
- add handling of a duplicate answer and notification about it.

Fix:
 - bug with incorrect display of the modal window if the question was not created.
 - bug with the checkbox when changing the value to correct input.

### CODE REVIEW

File: src/components/AddingAnswer/index.tsx [hunk 1]
```diff
@@ -90,7 +94,7 @@ export const AddingAnswer = ({
           onPress={onPressCorrectAnswerHandler}
           isChecked={isChecked}
           disabled={!!disabledCheckbox}
-          fillColor={Color.GreenLight}
+          fillColor={checkingForDuplicate ? Color.Red : Color.GreenLight}
```
Comment (antoniosk10) [inline src/components/AddingAnswer/index.tsx:97]: if your variable return boolean, its name should start from `is, are, has` and etc. https://www.syncfusion.com/blogs/post/10-javascript-naming-conventions-every-developer-should-know.aspx#:~:text=2.%20Naming%20Convention%20for%C2%A0Booleans

File: src/components/ui/ReadyStyles/CircleButton.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import {StyleSheet, TouchableOpacity} from 'react-native';
+import React from 'react';
+
+type CircleButtonPropsType = {
+  children: React.ReactNode;
+  onPress: () => void;
+  disabled: boolean;
+};
+
+export const CircleButton = ({children, onPress, disabled}: CircleButtonPropsType) => {
```
Comment (antoniosk10) [inline src/components/ui/ReadyStyles/CircleButton.tsx:10]: why is this called `CircleButton`? I just see `TouchableOpacity` wrapper with one style `opacity`

File: src/screens/CreateTest/CreateQuestion/CreateAnswer/index.tsx [hunk 3]
```diff
@@ -25,6 +28,7 @@ export const CreateAnswer = ({
   checkedCorrectOption,
 }: CreateAnswerPropsType) => {
   const disabledDeleteBtn = fields.length <= 2;
```
Comment (antoniosk10) [inline src/screens/CreateTest/CreateQuestion/CreateAnswer/index.tsx:30]: isDisabledDeleteBtn

File: src/screens/CreateTest/CreateQuestion/index.tsx [hunk 4]
```diff
@@ -62,31 +78,36 @@ export const CreateQuestion = ({
     correctAnswers: currentQuestion.content.correctAnswer,
   });
 
-  const onPressSaveQuestionHandler = (values: InputsFieldType) => {
+  const watchFieldArray = watch('options') || [];
+  const arrayOptions = watchFieldArray.map(el => el.option).filter(el => el !== '');
```
Comment (antoniosk10) [inline src/screens/CreateTest/CreateQuestion/index.tsx:82]: `const arrayOptions = watchFieldArray.filter(el => el.option !== '')`

Comment (antoniosk10) [inline src/screens/CreateTest/CreateQuestion/index.tsx:82]: useMemo

File: src/screens/CreateTest/CreateQuestion/index.tsx [hunk 5]
```diff
@@ -62,31 +78,36 @@ export const CreateQuestion = ({
...
+  const checkingForDuplicate = new Set(arrayOptions).size !== arrayOptions.length;
+
+  const nextQuestionPressed = useCallback(() => {
+    props.onPressCurrentQuestionPressed(props.isActiveTab + 1);
+    props.scrollRef.current?.scrollTo({x: 0, y: 0, animated: true});
+  }, [props]);
+
+  const onPressSaveQuestionHandler = async (values: InputsFieldType) => {
     const isTime = transformTime({
```
Comment (antoniosk10) [inline src/screens/CreateTest/CreateQuestion/index.tsx:91]: this vice versa, variable shouldn't start from `is`, change pls

File: src/screens/CreateTest/CreateQuestion/index.tsx [hunk 6]
```diff
@@ -62,31 +78,36 @@ export const CreateQuestion = ({
...
       format: 'onlySeconds',
       isMinutes: values.minutes,
       isSeconds: values.seconds,
     });
     const isOptions = transformFormatOptions(values.options);
```
Comment (antoniosk10) [inline src/screens/CreateTest/CreateQuestion/index.tsx:96]: the same, change name

File: src/screens/CreateTest/CreateQuestion/index.tsx [hunk 7]
```diff
@@ -192,7 +218,11 @@ export const CreateQuestion = ({
       </BlockBox>
       <BlockBox>
         <TextBox>Question difficulty</TextBox>
-        <SwitchSelectors type="level" onPress={selectQuestionDifficult} />
+        <SwitchSelectors
+          type="level"
```
Comment (antoniosk10) [inline src/screens/CreateTest/CreateQuestion/index.tsx:222]: use enum here

File: src/screens/CreateTest/QuestionsSettings/index.tsx [hunk 8]
```diff
@@ -35,22 +35,28 @@ export const QuestionsSettings = ({
   );
   const dispatch = useAppDispatch();
   const listQuestionsTabs = [...Array(route.params.numberQuestions)].map((el, i) => i);
+  const [isActiveTab, setIsActiveTab] = useState(0);
```
Comment (antoniosk10) [inline src/screens/CreateTest/QuestionsSettings/index.tsx:38]: activeTab

File: src/screens/CreateTest/CreateQuestion/index.tsx [hunk 9]
```diff
@@ -214,23 +244,43 @@ export const CreateQuestion = ({
...
+        <ViewDynamicFlex justifyC="flex-end" alignI="center" flexD="row">
+          <ContainerSaveButton>
+            <AppButton
+              title="Save question"
+              type="primary"
+              onPress={handleSubmit(onPressSaveQuestionHandler)}
+              disabled={checkingForDuplicate}
+            />
+          </ContainerSaveButton>
+          {props.isActiveTab + 1 === props.numberQuestions ? (
```
Comment (antoniosk10) [inline src/screens/CreateTest/CreateQuestion/index.tsx:265]: create varible for that calculation

File: src/screens/CreateTest/CreateQuestion/index.tsx [hunk 10]
```diff
@@ -214,23 +244,43 @@ export const CreateQuestion = ({
...
+            <AppButton
+              title="Exit"
+              type="primary"
```
Comment (antoniosk10) [inline src/screens/CreateTest/CreateQuestion/index.tsx:268]: create enum

**Review by antoniosk10 — APPROVED** [review]

**Review by KhaninPasha0488 — APPROVED** [review]

**Pull request merged**